packages/service-library/tests/rabbitmq/test_rabbitmq.py

```diff
@@ -303,6 +303,71 @@ async def _always_returning_fail(_: Any) -> bool:
     }
 
 
+@pytest.mark.parametrize("topics", _TOPICS)
+@pytest.mark.no_cleanup_check_rabbitmq_server_has_no_errors()
+async def test_publish_with_no_registered_subscriber(
+    on_message_spy: mock.Mock,
+    create_rabbitmq_client: Callable[[str], RabbitMQClient],
+    random_exchange_name: Callable[[], str],
+    random_rabbit_message: Callable[..., PytestRabbitMessage],
+    mocked_message_parser: mock.AsyncMock,
+    topics: list[str] | None,
+):
+    publisher = create_rabbitmq_client("publisher")
+    consumer = create_rabbitmq_client("consumer")
+
+    exchange_name = f"{random_exchange_name()}"
+
+    ttl_s: float = 0.1
+    topics_count: int = 1 if topics is None else len(topics)
+
+    async def _publish_random_message():
+        if topics is not None:
```

Comment:
TIP: you can redefined `topics = topics or []` to make sure it is always a list

Reply:
I could have written it better actually.